### ISSUE TITLE
BokehUserWarning: ColumnDataSource's columns must be of the same length" when plotting Paths but not Polygons

### ISSUE THREAD

**Issue by xaviernogueira** [holoviz/hvplot#1107]
#### ALL software version info
- `hvplot` version = '0.8.3'
- `geoviews` version = '1.9.6'
- `holoviews` version = '1.16.0'
- Python version = 3.10

#### Description of expected behavior and the observed behavior
When plotting a GeoDataFrame with MultiLines, if I attempt to provide an argument for `alpha` or `c` I get a BokehUserWarning: ColumnDataSource's columns must be of the same length" warning and don't receive any plot. **Oddly, I don't get this warning when I plot Polygons**.

For example:
```python
my_geo_df[['geometry', 'tp_conc']].hvplot(
    geo=True,
    crs=my_geo_df.crs.to_string(),
    c='tp_conc',
)
```
fails with this message:
`BokehUserWarning: ColumnDataSource's columns must be of the same length. Current lengths: ('color', 99), ('tp_conc', 3755), ('xs', 3755), ('ys', 3755)`

OR 
```python
my_geo_df[['geometry', 'tp_conc']].hvplot(
    geo=True,
    crs=my_geo_df.crs.to_string(),
    alpha=gv.dim('tp_conc'),
```

fails with this message:
`BokehUserWarning: ColumnDataSource's columns must be of the same length. Current lengths: ('alpha', 99), ('tp_conc', 3755), ('xs', 3755), ('ys', 3755)`

**Why could the dimension created by geo/holoviews for `alpha` or `color` be way smaller than the other dimensions?**

Thanks

**Comment by ahuang11** [comment]
Do you have some data that we can reproduce with?

**Comment by xaviernogueira** [comment]
Yes. Try https://github.com/WikiWatershed/pollution-assessment/blob/main/stage2/data_output/reach_concs_gdf.parquet

It's a GeoParquet. This is the code I am testing on:

```python
reach_concs_gdf = gpd.read_parquet(
    PA2_DATA_DIR /'reach_concs_gdf.parquet'
)[:100]
```

**Comment by prusswan** [comment]
After running into other glitches involving plots, I suspect this could be related to NaN handling e.g. `np.isfinite` is one of the operations that could alter the data size if NaN values are present. One of the errors look like:


```
File ~/anaconda3/envs/py311/lib/python3.11/site-packages/geoviews/operation/projection.py:189, in <dictcomp>(.0)
    187 mask = np.isfinite(coordinates[:, 0])
    188 dims = [d for d in element.dimensions() if d not in (xdim, ydim)]
--> 189 new_data = {k: v[mask] for k, v in element.columns(dims).items()}
    190 new_data[xdim.name] = coordinates[mask, 0]
    191 new_data[ydim.name] = coordinates[mask, 1]

IndexError: boolean index did not match indexed array along axis 0; size of axis is 48 but size of corresponding boolean axis is 791
```
Even if there was no error on line 189, if NaN values are present in `coordinates`, `new_data` would end up with reduced size compared to the input data.

Not sure if this is an issue with geoviews or how the other libraries are interacting with it. In view of this, I would suggest to look out for NaN values in case of unexpected changes to data dimensions.

possibly related: holoviz/hvplot#792, holoviz/holoviews#4862
